src/main/java/de/dennisguse/opentracks/io/file/importer/ImportService.java

```diff
@@ -0,0 +1,90 @@
+package de.dennisguse.opentracks.io.file.importer;
+
+import android.app.job.JobService;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.ResultReceiver;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.JobIntentService;
+import androidx.documentfile.provider.DocumentFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.io.file.TrackFileFormat;
+import de.dennisguse.opentracks.util.FileUtils;
+
+public class ImportService extends JobIntentService {
+
+    private static final String TAG = ImportService.class.getSimpleName();
+
+    private static final int JOB_ID = 2;
+
+    private static final String EXTRA_RECEIVER = "extra_receiver";
+    private static final String EXTRA_URI = "extra_uri";
+
+    private ResultReceiver resultReceiver;
+
+    public static void enqueue(Context context, ImportServiceResultReceiver receiver, Uri uri) {
```

Comment:
Can we change to pass a DocumentFile instead of Uri?

Reply:
No, we can't. DocumentFile doesn't implement Parcelable so we cannot pass it through Extra.